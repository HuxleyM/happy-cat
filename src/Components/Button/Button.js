import React from 'react';
import Styles from'./Button.module.css';



function Button({text, state}) {
    return (
    <button disabled={state !== 'success'} className={`${Styles.mainActionButton} ${Styles[state]}`}>{text}</button>
    )
}

export default Button;
