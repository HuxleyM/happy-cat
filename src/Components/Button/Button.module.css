.mainActionButton{
    color: black;
    height:54px;
    width:188px;
    border-radius:10px;
    cursor: pointer;
    border-style: none;
}

.error{
    background: #EB6953;
}

.success{
    background: #81994D;
}

.incomplete{
    background: #C4C4C4;
}