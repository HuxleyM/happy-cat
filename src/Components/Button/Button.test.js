import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';


const BUTTON_TEXT = 'jungle is massive'


describe('Button', () => {
  const wrapped = shallow(<Button text={BUTTON_TEXT} state='error'></Button>);

  it('should render the Button with text', () => {  
      expect(wrapped.text().includes(BUTTON_TEXT)).toBe(true)
  });


    it('Button is passed success should be enabled and green', () => {
        const wrapped = shallow(<Button text={BUTTON_TEXT} state='success'></Button>);
        expect(wrapped).toMatchSnapshot();
    })

    it('Button is passed error should be disabled and red', () => {
        const wrapped = shallow(<Button text={BUTTON_TEXT} state='error'></Button>);
        expect(wrapped).toMatchSnapshot();
    })

    it('Button is passed incomplete should be disabled and grey', () => {
        const wrapped = shallow(<Button text={BUTTON_TEXT} state='incomplete'></Button>);
        expect(wrapped).toMatchSnapshot();
    })



});


