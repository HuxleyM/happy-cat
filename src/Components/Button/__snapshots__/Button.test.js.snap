// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button is passed error should be disabled and red 1`] = `
<button
  className="mainActionButton error"
  disabled={true}
>
  jungle is massive
</button>
`;

exports[`Button Button is passed incomplete should be disabled and grey 1`] = `
<button
  className="mainActionButton incomplete"
  disabled={true}
>
  jungle is massive
</button>
`;

exports[`Button Button is passed success should be enabled and green 1`] = `
<button
  className="mainActionButton success"
  disabled={false}
>
  jungle is massive
</button>
`;
